### PR TITLE
Round Tests

### LINKED ISSUE #5148: Round preventing Pokemon to make a Move
### Description

If the AI manages to setup a double Round, it can change the Turnorder dynamically. This seems to create some issues and can make one of the Players Pokemon unable to move. Here is a clip for reference where Leavanny was not allowed to get its move off. (It's not because of Throat Spray, I tested further with this)
Note: The Partner switching out is necessary to recreate this. For some reason it has to be the Pokemon in the right switching out to trigger this Bug.

https://github.com/user-attachments/assets/aeb9bb9b-528e-43ec-b849-f9615755e751

### Version

1.9.0 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

._.terra._.

### PR DESCRIPTION
**Pull request by hedara90**
Tests for the move Round.

## Description

Tests that other battlers that selected Round moves immediately after the first user of Round, both allies and foes.
Tests that the turn order doesn't change otherwise (no exhaustive)
Tests that the other usages of Round has doubled Base Power.
Tests that #5148 hasn't been fixed.

## Feature(s) this PR does NOT handle:

This PR does not fix #5148, it simply implements tests for the move and the bugged behavior.

## **Discord contact info**

hedara

### CODE REVIEW

File: test/battle/move_effect/round.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+{
+    GIVEN {
+        ASSUME(gItemsInfo[ITEM_LAGGING_TAIL].holdEffect == HOLD_EFFECT_LAGGING_TAIL);
+        ASSUME(gMovesInfo[MOVE_IRON_HEAD].additionalEffects[0].moveEffect == MOVE_EFFECT_FLINCH);
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WYNAUT);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_LAGGING_TAIL); }
+    } WHEN {
+        TURN { MOVE(playerRight, MOVE_CELEBRATE); MOVE(opponentLeft, MOVE_ROUND, target: playerLeft); MOVE(playerLeft, MOVE_IRON_HEAD, target: opponentRight); MOVE(opponentRight, MOVE_ROUND, target: playerLeft); }
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/round.c:19]:
```suggestion
        TURN {
            MOVE(playerRight, MOVE_CELEBRATE);
            MOVE(opponentLeft, MOVE_ROUND, target: playerLeft);
            MOVE(playerLeft, MOVE_IRON_HEAD, target: opponentRight);
            MOVE(opponentRight, MOVE_ROUND, target: playerLeft);
        }
```

File: test/battle/move_effect/round.c [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+DOUBLE_BATTLE_TEST("Round usages beyond the first one has double base power")
+{
+    s16 damage[2];
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WYNAUT);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponentLeft, MOVE_ROUND, target: playerLeft); MOVE(opponentRight, MOVE_ROUND, target: playerLeft); }
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/round.c:36]:
```suggestion
        TURN {
            MOVE(opponentLeft, MOVE_ROUND, target: playerLeft);
            MOVE(opponentRight, MOVE_ROUND, target: playerLeft);
        }
```

File: test/battle/move_effect/round.c [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+DOUBLE_BATTLE_TEST("Round still preserves the turn order outside of the other Round users moving immediately")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WYNAUT);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponentLeft, MOVE_ROUND, target: playerLeft); MOVE(playerRight, MOVE_CELEBRATE); MOVE(playerLeft, MOVE_CELEBRATE); MOVE(opponentRight, MOVE_ROUND, target: playerLeft); }
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/round.c:53]:
```suggestion
        TURN {
            MOVE(opponentLeft, MOVE_ROUND, target: playerLeft);
            MOVE(playerRight, MOVE_CELEBRATE);
            MOVE(playerLeft, MOVE_CELEBRATE);
            MOVE(opponentRight, MOVE_ROUND, target: playerLeft); }
```

File: test/battle/move_effect/round.c [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+{
+    KNOWN_FAILING; // #5148
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WYNAUT);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { SWITCH(playerRight, 2); MOVE(opponentLeft, MOVE_ROUND, target: playerLeft); MOVE(playerLeft, MOVE_CELEBRATE); MOVE(opponentRight, MOVE_ROUND, target: playerLeft); }
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/round.c:71]:
```suggestion
        TURN {
            SWITCH(playerRight, 2);
            MOVE(opponentLeft, MOVE_ROUND, target: playerLeft);
            MOVE(playerLeft, MOVE_CELEBRATE);
            MOVE(opponentRight, MOVE_ROUND, target: playerLeft);
        }
```

File: test/battle/move_effect/round.c [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WYNAUT);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_LAGGING_TAIL); }
+    } WHEN {
+        TURN { MOVE(opponentLeft, MOVE_CELEBRATE); MOVE(playerRight, MOVE_ROUND, target: opponentLeft); MOVE(playerLeft, MOVE_CELEBRATE, target: opponentRight); MOVE(opponentRight, MOVE_ROUND, target: playerLeft); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_ROUND, playerRight);
+        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_CELEBRATE, playerLeft);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_ROUND, opponentRight);
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/round.c:92]:
```suggestion
        ANIMATION(ANIM_TYPE_MOVE, MOVE_CELEBRATE, opponentLeft);
        ANIMATION(ANIM_TYPE_MOVE, MOVE_ROUND, playerRight);
        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_CELEBRATE, playerLeft);
        ANIMATION(ANIM_TYPE_MOVE, MOVE_ROUND, opponentRight);
        ANIMATION(ANIM_TYPE_MOVE, MOVE_CELEBRATE, playerLeft);
```

File: test/battle/move_effect/round.c [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WYNAUT);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_LAGGING_TAIL); }
+    } WHEN {
+        TURN { MOVE(playerRight, MOVE_CELEBRATE); MOVE(opponentLeft, MOVE_ROUND, target: playerLeft); MOVE(playerLeft, MOVE_IRON_HEAD, target: opponentRight); MOVE(opponentRight, MOVE_ROUND, target: playerLeft); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_ROUND, opponentLeft);
+        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_IRON_HEAD, playerLeft);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_ROUND, opponentRight);
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/round.c:23]:
```suggestion
        ANIMATION(ANIM_TYPE_MOVE, MOVE_CELEBRATE, playerRight);
        ANIMATION(ANIM_TYPE_MOVE, MOVE_ROUND, opponentLeft);
        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_IRON_HEAD, playerLeft);
        ANIMATION(ANIM_TYPE_MOVE, MOVE_ROUND, opponentRight);
        ANIMATION(ANIM_TYPE_MOVE, MOVE_IRON_HEAD, playerLeft);
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Comment by AsparagusEduardo** [comment]
Some readability suggestions, but also recommends checking all 4 move animations just in case

**Comment by hedara90** [comment]
Applied suggestions

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**